app/src/componentsV2/BottomSheet/context/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useState, useContext } from "react";
+import { BottomSheetPlacement } from "../types";
+
+interface BottomSheetContextProps {
+  isBottomSheetOpen: boolean;
+  sheetPlacement: BottomSheetPlacement;
+  toggleBottomSheet: () => void;
+  changeSheetPlacement: (placement: BottomSheetPlacement) => void;
+}
+
+const BottomSheetContext = createContext<BottomSheetContextProps | undefined>(undefined);
+
+export const BottomSheetProvider: React.FC<{ children: React.ReactNode; defaultPlacement: BottomSheetPlacement }> = ({
+  children,
+  defaultPlacement,
+}) => {
+  const [isBottomSheetOpen, setIsBottomSheetOpen] = useState(false);
+  const [sheetPlacement, setSheetPlacement] = useState(defaultPlacement);
+
+  const toggleBottomSheet = () => setIsBottomSheetOpen(!isBottomSheetOpen);
+  const changeSheetPlacement = (placement: BottomSheetPlacement) => {
```

Comment:
This should be `toggleSheetPlacement`. And why are we checking the passed value as argument and toggling it? Shouldn't we check the state value `sheetPlacement` and then toggle it 🤔